### PR TITLE
LUCENE-10107

### PR DESCRIPTION
**Pull request by madrob**
https://issues.apache.org/jira/browse/LUCENE-10107

### CODE REVIEW

File: dev-tools/scripts/smokeTestRelease.py
Comment (dweiss) [inline dev-tools/scripts/smokeTestRelease.py:1225]: If you're using the short hash version then the previous line was correct (+ instead of fixed-character count 11)?

Comment (madrob) [reply dev-tools/scripts/smokeTestRelease.py:1225]: This regex extracts the hash from the filename, which still has a full sha. It looks something like `$TMPDIR/releases-8.10/lucene-solr-8.10.0-RC3-revb473a4f88588bb1e32cff27aaee05380c42f64f1`

Comment (dweiss) [reply dev-tools/scripts/smokeTestRelease.py:1225]: Ah, ok, clear, thanks!

File: lucene/common-build.xml
Comment (dweiss) [inline lucene/common-build.xml:767]: I think it could also be a more verbose command like this one (note the extended abbreviated hash syntax):
```
git show --pretty=oneline --abbrev-commit --abbrev=15 --no-patch HEAD
```

**Comment by madrob** [comment]
This worked for me on macOS but I'd like somebody else to try this with Windows or Linux before we merge?

**Review by thelabdude — APPROVED** [review]
Tried these changes out in the 8.10 RC build checkout I have locally and works well but on Mac too.

File: dev-tools/scripts/smokeTestRelease.py
Comment (janhoy) [inline dev-tools/scripts/smokeTestRelease.py:173]: Will the MANIFEST line endings always be `CRLF` ? According to the spec it could be `CR LF | LF | CR`...

Comment (madrob) [reply dev-tools/scripts/smokeTestRelease.py:173]: I'm not sure if CRLF on my machine was a Mac setting or something that ant or java tooling was doing. Hence the request for Windows/Linux testing help. But I chatted with Tim and confirmed that at least _his_ machine was also producing CRLF, so this wouldn't be holding up the release. I guess the correct expression would be `(\r\n?|\r) ` and then would have to do something more intelligent with the line removal too.

**Pull request merged**
